client/signup/reskinned-processing-screen/index.jsx

```diff
@@ -58,11 +59,11 @@ export default function ReskinnedProcessingScreen( props ) {
 	const shouldShowNewSpinner =
 		isDestinationSetupSiteFlow || flowsWithDesignPicker.includes( flowName );
 
-	const duration = flowsWithDesignPicker.includes( flowName )
-		? HEADSTART_DURATION_IN_MS
-		: DURATION_IN_MS;
 	const [ currentStep, setCurrentStep ] = useState( 0 );
 
+	const defaultDuration = DURATION_IN_MS / totalSteps;
+	const duration = steps.current[ currentStep ]?.duration || defaultDuration;
```

Comment:
On this line I added the null check `?`

Reply:
Sounds strange as the interval won't triggered when the progress is completed 🤔